crates/edr_provider/src/lib.rs

```diff
@@ -100,224 +105,251 @@ impl Provider {
-                .and_then(to_json)
-        }
-        MethodInvocation::GetTransactionByBlockNumberAndIndex(block_spec, index) => {
-            eth::handle_get_transaction_by_block_spec_and_index(data, block_spec, index)
-                .and_then(to_json)
-        }
-        MethodInvocation::GetTransactionByHash(transaction_hash) => {
-            eth::handle_get_transaction_by_hash(data, transaction_hash).and_then(to_json)
-        }
-        MethodInvocation::GetTransactionCount(address, block_spec) => {
-            eth::handle_get_transaction_count_request(data, address, block_spec).and_then(to_json)
-        }
-        MethodInvocation::GetTransactionReceipt(transaction_hash) => {
-            eth::handle_get_transaction_receipt(data, transaction_hash).and_then(to_json)
-        }
-        MethodInvocation::Mining(()) => Err(ProviderError::Unimplemented("Mining".to_string())),
-        MethodInvocation::NetListening(()) => eth::handle_net_listening_request().and_then(to_json),
-        MethodInvocation::NetPeerCount(()) => {
-            eth::handle_net_peer_count_request().and_then(to_json)
-        }
-        MethodInvocation::NetVersion(()) => eth::handle_net_version_request(data).and_then(to_json),
-        MethodInvocation::NewBlockFilter(()) => {
-            Err(ProviderError::Unimplemented("NewBlockFilter".to_string()))
-        }
-        MethodInvocation::NewFilter(_) => {
-            Err(ProviderError::Unimplemented("NewFilter".to_string()))
-        }
-        MethodInvocation::NewPendingTransactionFilter(()) => {
-            eth::handle_new_pending_transaction_filter_request(data).and_then(to_json)
-        }
-        MethodInvocation::PendingTransactions(()) => {
-            eth::handle_pending_transactions(data).and_then(to_json)
-        }
-        MethodInvocation::SendRawTransaction(raw_transaction) => {
-            eth::handle_send_raw_transaction_request(data, raw_transaction).and_then(to_json)
-        }
-        MethodInvocation::SendTransaction(transaction_request) => {
-            eth::handle_send_transaction_request(data, transaction_request).and_then(to_json)
-        }
-        MethodInvocation::Sign(message, address) => {
-            eth::handle_sign_request(data, message, address).and_then(to_json)
-        }
-        MethodInvocation::SignTypedDataV4(_, _) => {
-            Err(ProviderError::Unimplemented("SignTypedDataV4".to_string()))
-        }
-        MethodInvocation::Subscribe(_) => {
-            Err(ProviderError::Unimplemented("Subscribe".to_string()))
-        }
-        MethodInvocation::Syncing(()) => Err(ProviderError::Unimplemented("Syncing".to_string())),
-        MethodInvocation::UninstallFilter(filter_id) => {
-            eth::handle_uninstall_filter_request(data, filter_id).and_then(to_json)
-        }
-        MethodInvocation::Unsubscribe(filter_id) => {
-            eth::handle_unsubscribe_request(data, filter_id).and_then(to_json)
-        }
+    fn handle_single_request(
```

Comment:
Were there any significant changes to this function? Or is it just a formatting that triggered the git changes?

Reply:
I turned it from a "free" function to a member function, as one of the requests needs access to the `IntervalMiner`. This was the best way to model that